DEPLOYMENT.md

```diff
@@ -0,0 +1,79 @@
+
+The workflow requires two inputs: `environment` and `tag`. `environment` must be either `staging` or `production` and the `tag` must be a valid tag that exists on the ghcr.io/wordpress/openverse-frontend image. The `tag` input is validated against the available list of tags and the workflow will fail if the tag is determined not to exist in the image repository.
+
+After validating the tag and that the dispatcher is authorised, the Rollback workflow follows the deployment process described above, except that it skips the Docker image build and goes straight to step 2. Because of this difference, the rollback workflow generally takes less than 10 minutes.
+
+## Future end-to-end test integration
+
+In the future we hope to integrate end-to-end tests into this workflow and to tighten the staging to production pipeline. Essentially, we would like the deployment pipeline to go something like this:
+
+1. Merge PR to `main`.
+2. Wait for staging to deploy.
+3. Playwright tests automatically run against the new staging environment.
+4. While Playwright tests are running, the person who merged the PR should manually verify that their change behaves as expected.
+5. After the Playwright tests are confirmed to have passed, unlock a "deploy to production" button somewhere that pushes the image currently deployed to staging into the production task definition.
+
+Following this process would mean that production gets deployed every time code is merged to `main`. It reduces the turn around for new features and bug fixes because they don't have to wait for a new release to be manually created.
+
+We could more or less follow the process as described today just by creating a new release every time we've merged to staging and verified our changes, but there are a couple missing pieces that would need to be implemented to make this process easy to follow, to varying degrees of difficulty:
+
+1. We need a GitHub workflow that runs the Playwright tests against the staging deployment automatically after staging is deployed. This will require some Playwright configuration tweaks and increased test timeouts due to network latency.
+2. We would need to create a queue so that new staging deployments do not occur until the current staging deployment is live on production. This could be implemented by polling the `/version.json` endpoints of both environments and only liberating a new staging deployment for new merges once the endpoints both report the same `release` property.
+3. We would need to update the production deployment workflow to not create a new Docker build and instead just pull the current tag from the staging `/version.json` endpoint.
+
+Note that this workflow does make the process of merging a PR heavier than it is today. This comes with some benefits:
+
+1. PRs will likely be more carefully tested as the consequence of merging them is higher and more immediately consequential.
+2. Bugs will be fixed in production more quickly.
```

Comment:
This **Future end-to-end test integration** is an exciting plan ✨ and I immediately thought about these points while reading. When the time comes, we would need to really compromise in not merging by pure urgency and include tests with the changes, or I guess we can also define very clearly the exceptions to these constraints.

Reply:
I'm interested in ways in which we can hold ourselves accountable for better testing. Coverage data could be one way but I'm sure there are other creative ones.